### ISSUE TITLE
[Suggestion] Improvements for a better macOS version

### ISSUE THREAD

**Issue by brunocastello** [sarah-walker-pcem/pcem#131]
Hi, I'd like to put forward some suggestions for the macOS version of PCem. I managed to compile the current code for the M1 Mac and I am testing it as we speak. Some things could be improved for an easier experience for macOS users:

1) **Script to create a macOS bundle app.** @kyr0 and @almeath have worked in the past together to put some stuff like this (https://github.com/PCemOnMac/bundler) and these two scripts (https://github.com/PCemOnMac/MacOS-Finder-Convert-Folder-To-CD-Image and https://github.com/PCemOnMac/MacOS-Finder-Convert-Folder-To-Floppy-Disk) which help to create CD and Floppy images.

2) **Default folder for configs, roms and screenshots (~/PCem)**. Currently the binary runs if I have these folders next to it. @PCemOnMac builds were pointing to ~/PCem inside the users home folder, but some users may want to run from external storage (when some Macs have 128 or 256GB only or they want to use the machines on another computer). A pop-up on first run of PCem asking where the user wants the default folder to be created, is the best UI/UX solution. DOSBox-X does that.

By doing that (and also having the same second suggestion implemented on other platforms), I believe that PCem v18 could be a great release across all platforms. The switch to CMake/Ninja to compile and build was impressive.

**Comment by michael-manley** [comment]
I am going to make #131 now the "Mac Megaticket" :)

I will mention I do own a Mac so I can test a few dev issues but my macs are a bit older. I more or less do Linux and Windows testing.

Item 1 i think can be done by CMake, ill look into creating a bundle.
Item 2 pcem.cfg has a part where it saves directory paths
![image](https://user-images.githubusercontent.com/291225/163098884-c997d0fe-1575-4a3b-bb9b-a05b0136cfa2.png)
What I will change for UNIX OS'es is read pcem.cfg from the binary first, if does not exists read in ~/.pcem

**Comment by brunocastello** [comment]
Thanks! I have tested the latest binary I successfully built some days ago. I tried to run a Win98 PII 233 machine with Voodoo 2 and later with a Voodoo3 3000. Games tested were Need For Speed II SE and Grand Prix 3, respectively. Grand Prix 3 has support for Voodoo3 cards, but NFS2SE failed to detect 3Dfx support from that same card during the installation.

Both failed to launch - I guess that the voodoo emulation is still broken for macOS users? It was working well between 2019 and 2020, before the first attempts were made to adjust the code to compile for M1 Macs as well.

Other than that, the machine runs well. One thing that caught my attention was the performance. I owned a 2013 MBP "13 Intel i7 16GB and 1TB SSD in 2020, and the performance with my current 2020 M1 MacBook Air is quite the same. I was expecting the M1 to be faster (for obvious reasons), and to allow me to use the Pentium II machines.

The 2013 MBP best maximum performance was achieved with a Pentium 133 MMX and FIC-VA-503+ in 2020. I haven't tested that configuration yet here on my M1.

Other things that were already known from previous PCem builds for macOS:

- A small black box shows and sticks for the entire usage of the machine in the middle of my desktop. Does not disturb the usage of the machine though;

- Since the very first builds for macOS that @kyr0 and @almeath built on @PCemOnMac, the app still crashes on exit;

- After the shutdown, the machine window still sticks behind the main PCem window, in my case showing the windows 98 shutdown screen (not the black one at the end).

- During my tests I noticed that the build was pretty unstable at times - crashing back to my desktop during its usage for different reasons (I should have tried to log them so I could explain what really happened).

I hope that these feedbacks are somewhat useful... I'm not exactly a programmer, I am an UI/UX Designer and I can only test (hey, it's the fun part of the task) 🤷🏻‍♂️

**Comment by brunocastello** [comment]
> Item 2 pcem.cfg has a part where it saves directory paths ![image](https://user-images.githubusercontent.com/291225/163098884-c997d0fe-1575-4a3b-bb9b-a05b0136cfa2.png) What I will change for UNIX OS'es is read pcem.cfg from the binary first, if does not exists read in ~/.pcem

Yeah I investigated it some time after the first run of the binary and noticed that file. I edited it and worked, pcem was looking for these files exactly where I modified in pcem.cfg, so this is reading the cfg file well.

**Comment by brunocastello** [comment]
As of now, with the current source code, I can't compile on M1 Mac due to some cmake errors.
<img width="540" alt="Screen Shot 2022-04-30 at 2 14 55 PM" src="https://user-images.githubusercontent.com/23453202/166105105-b1d8567c-3589-4fd2-8a9c-82570050df79.png">

This when I run `cmake -G "Ninja" -DCMAKE_BUILD_TYPE=Release -DUSE_NETWORKING=ON && ninja` on project directory (/pcem)

And this when I go to /pcem/src and attempt to run the compilation again:
<img width="570" alt="Screen Shot 2022-04-30 at 2 16 08 PM" src="https://user-images.githubusercontent.com/23453202/166105161-90b7e36c-903b-4e32-a3f3-50c1ae3e2ed3.png">

**Comment by brunocastello** [comment]
@michael-manley did you have some luck looking into this issue?

**Comment by ylluminate** [comment]
Any updates on this situation? Would very much like to run this on a few Apple Silicon machines in the near term. Happy to test, but the status doesn't seem quite clear yet.

**Comment by embusalacchi** [comment]
I am trying to compline on an M1 Max and I'm seeing the error below.  I would love to get this running on Apple Silicon and would love to help test if I can get it to compile.  Any help would be appreciated!

```
[build] ld: symbol(s) not found for architecture arm64
[build] clang: error: linker command failed with exit code 1 (use -v to see invocation)
[build] ninja: build stopped: subcommand failed.
```

**Follow Up** - This command was able to build pcem successfully on M1 Mac:

```
cmake -G "Ninja" -DCMAKE_BUILD_TYPE=Release -DUSE_NETWORKING=ON -DCMAKE_OSX_ARCHITECTURES=arm64 .
ninja
```

**Comment by RichCini** [comment]
Adding something new to this...trying to compile under OSX Sequoia 15 (Intel) and it barfs when compiling soundopenal.c, not being able to locate OpenAL/al.h. The command line tools and XCode are up-to-date.

`/Users/racini/src/pcem-dev/src/sound/soundopenal.c:9:10: note: did not find header 'al.h' in framework 'OpenAL' (loaded from '/Users/racini/Library/Frameworks')`

It points to the user library folder, but the headers are buried deep within the system library folder where they're installed by XCode. Is there an override/switch I need to force it to look in the right space?

Thanks!
Rich
